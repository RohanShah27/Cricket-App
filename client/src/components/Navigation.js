import React, { Component } from "react";
import { getGlobalSearchResult } from "../actions/Search";
import "../styles/navbar.css";
import { Link } from "react-router-dom";
import { connect } from "react-redux";

export class Navigation extends Component {
  constructor(props) {
    super(props);
  }
  state = {
    search_term: ""
  };
  removeToken = () => {
    localStorage.removeItem("token");
    this.forceUpdate();
  };
  getResult = () => {
    let search_term = {
      search_term: this.state.search_term
    };
    this.props.getGlobalSearchResult(search_term);
  };

  OnChange = event => {
    this.setState({ [event.target.name]: event.target.value });
    let search_term = {
      search_term: this.state.search_term
    };
    this.props.getGlobalSearchResult(search_term);
  };

  render() {
    console.log(this.props);
    return (
      <div className="nav-container">
        <div className="nav-header">
          <a className="logo">
            <Link to="/">Crickstrait</Link>
          </a>
          {this.props.gender == "male" ? (
            <a className="options" style={{ color: "#c2c2c2c2" }}>
              Men
            </a>
          ) : (
            <a className="options" onClick={this.props.menState}>
              Men
            </a>
          )}
          {this.props.gender == "female" ? (
            <a className="options" style={{ color: "#c2c2c2c2" }}>
              Women
            </a>
          ) : (
            <a className="options" onClick={this.props.femaleState}>
              Women
            </a>
          )}

          <input className="menu-btn" type="checkbox" id="menu-btn" />
          <label className="menu-icon" for="menu-btn">
            <span className="navicon"></span>
          </label>

          <ul className="menu">
            <li>
              <Link to="/rankings">Rankings</Link>
            </li>
            <li>
              <Link to="/matches">Matches</Link>
            </li>
            <li>
              <Link to="/teams">Teams</Link>
            </li>
            <li>
              <Link to="/players">Players</Link>
            </li>
            <li style={{ textAlign: "center" }}>
              <input
                className="nav-search-bar"
                type="text"
                name="search_term"
                value={this.state.search_term}
                onChange={this.OnChange}
              />
              <button className="nav-search-button" onClick={this.getResult}>
                <i className="fa fa-search"></i>
              </button>
              {this.state.search_term == "" ? null : (
                <div className="search-result">
                  {this.props.search_result.length != 0
                    ? this.props.search_result.player.length != 0
                      ? this.props.search_result.player.map(player => (
                          <span className="search_result_data">
                            <Link
                              to={{
                                pathname: "/playerprofile/" + player.player_id,
                                state: { player }
                              }}
                            >
                              <p>{player.player_name}</p>
                            </Link>
                          </span>
                        ))
                      : "No Player Found"
                    : null}
                  {this.props.search_result.length != 0
                    ? this.props.search_result.team.length != 0
                      ? this.props.search_result.team.map(team => (
                          <span className="search_result_data">
                            <Link
                              to={{
                                pathname: "/viewteam/" + team.team_id,
                                state: { team }
                              }}
                            >
                              <p>{team.team_name}</p>
                            </Link>
                          </span>
                        ))
                      : "No Team Found"
                    : null}
                </div>
              )}
            </li>
            {localStorage.getItem("token") ? (
              <li>
                <Link to="/" onClick={this.removeToken}>
                  Logout
                </Link>
              </li>
            ) : (
              <li>
                <Link to="/login">Login</Link>
              </li>
            )}
          </ul>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  search_result: state.searchReducer.search_result
});
export default connect(
  mapStateToProps,
  { getGlobalSearchResult }
)(Navigation);
