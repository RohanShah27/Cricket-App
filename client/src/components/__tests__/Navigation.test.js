import React from "react";
import { shallow, mount } from "enzyme";
import { Navigation } from "../Navigation";

let search_result = {
  player: [
    {
      player_id: 4,
      player_name: "Test name"
    }
  ],
  team: [
    {
      team_id: 4,
      team_name: "Team Test Name"
    }
  ]
};
const menState = jest.fn();
const femaleState = jest.fn();
const navigation = jest.fn();
const getGlobalSearchResult = jest.fn();
const wrapper = shallow(
  <Navigation
    navigation={navigation}
    menState={menState}
    femaleState={femaleState}
    getGlobalSearchResult={getGlobalSearchResult}
    search_result={search_result}
  />
);

describe("Testing Footer Component", () => {
  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("shoudl check for input value", () => {
    const e = {
      target: {
        name: "search_term",
        value: "virat"
      }
    };
    expect(wrapper.state().search_term).toBe("");
    wrapper.instance().OnChange(e);
    expect(wrapper.state().search_result).toBe(undefined);
    expect(getGlobalSearchResult).toBeCalledTimes(1);
  });
  it("should map players from search result", () => {
    expect(wrapper.find("#searchPlayer0").props().children).toStrictEqual([
      " ",
      search_result.player[0].player_name
    ]);
  });
  it("should map teams from search result", () => {
    expect(wrapper.find("#teamName0").props().children).toBe(
      search_result.team[0].team_name
    );
  });
  it("should check if Link is present", () => {
    expect(wrapper.find("Link").exists()).toBeTruthy();
  });
  it("should check for brand name and links", () => {
    expect(wrapper.find("#Brand").text()).toBe("Crickstrait");
  });
  it("should check for links", () => {
    expect(wrapper.find("#Rankings").text()).toBe("Rankings");
    expect(wrapper.find("#teams").text()).toBe("Teams");
    expect(wrapper.find("#matches").text()).toBe("Matches");
    expect(wrapper.find("#players").text()).toBe("Players");
  });
  it("should check for search bar", () => {
    expect(wrapper.find("#globalSearchBar").length).toBe(1);
    expect(wrapper.find("#globalSearchBar").props().placeholder).toBe(
      undefined
    );
  });
  it("should check for button", () => {
    expect(wrapper.find("#searchButton").length).toBe(1);
    // expect(wrapper.find("#searchButton").simulate());
    let getResult = jest.spyOn(wrapper.instance(), "getResult");
    expect(getResult).toBeCalledTimes(0);
  });
  it("should check for men state function", () => {
    expect(wrapper.find("#menState").text()).toBe("Men");
  });
  it("should check for feamale state ", () => {
    expect(wrapper.find("#femaleState").text()).toBe("Women");
  });
  it("should check for presense of remove token function", () => {
    let removeToken = jest.spyOn(wrapper.instance(), "removeToken");
    expect(removeToken).toBeTruthy();
  });
  it("no of div tags", () => {
    expect(wrapper.find("div").length).toBe(3);
  });

  it("no of p tags", () => {
    expect(wrapper.find("p").length).toBe(2);
  });

  it("no of i tags", () => {
    expect(wrapper.find("i").length).toBe(1);
  });

  it("no of div tag className", () => {
    expect(
      wrapper
        .find("div")
        .at(0)
        .props().className
    ).toBe("nav-container");
  });

  it("no of div tag className", () => {
    expect(
      wrapper
        .find("div")
        .at(0)
        .props().className
    ).toBe("nav-container");
  });

  it("simulate search button", () => {
    expect(
      wrapper
        .find("button")
        .at(0)
        .simulate("click")
    );
  });
});
