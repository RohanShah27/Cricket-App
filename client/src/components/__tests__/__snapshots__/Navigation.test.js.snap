// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer Component should mount the component 1`] = `
<div
  className="nav-container"
>
  <div
    className="nav-header"
  >
    <a
      className="logo"
      id="Brand"
    >
      <Link
        to="/"
      >
        Crickstrait
      </Link>
    </a>
    <a
      className="options"
      id="menState"
      onClick={[MockFunction]}
      style={
        Object {
          "marginRight": "5px",
        }
      }
    >
      Men
    </a>
    <a
      className="options"
      id="femaleState"
      onClick={[MockFunction]}
    >
      Women
    </a>
    <input
      className="menu-btn"
      id="menu-btn"
      type="checkbox"
    />
    <label
      className="menu-icon"
      for="menu-btn"
    >
      <span
        className="navicon"
      />
    </label>
    <ul
      className="menu"
    >
      <li
        id="Rankings"
      >
        <Link
          to="/rankings"
        >
          Rankings
        </Link>
      </li>
      <li
        id="matches"
      >
        <Link
          to="/matches"
        >
          Matches
        </Link>
      </li>
      <li
        id="teams"
      >
        <Link
          to="/teams"
        >
          Teams
        </Link>
      </li>
      <li
        id="players"
      >
        <Link
          to="/Players"
        >
          Players
        </Link>
      </li>
      <li
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <input
          className="nav-search-bar"
          id="globalSearchBar"
          name="search_term"
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          className="nav-search-button"
          id="searchButton"
          onClick={[Function]}
        >
          <i
            className="fa fa-search"
          />
        </button>
      </li>
      <li>
        <Link
          to="/login"
        >
          Login
        </Link>
      </li>
    </ul>
  </div>
</div>
`;
